core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaStoreMessageHandler.java

```diff
@@ -84,17 +118,85 @@ public void handleUpdate(SchemaRegistryKey key,
                            TopicPartition tp,
                            long offset,
                            long timestamp) {
+    if (key.getKeyType() != SchemaRegistryKeyType.SCHEMA && value == null) {
+      // ignore non-schema tombstone
+      return;
+    }
+
     if (key.getKeyType() == SchemaRegistryKeyType.SCHEMA) {
       handleSchemaUpdate((SchemaKey) key,
           (SchemaValue) value,
           (SchemaValue) oldValue);
-    } else if (value == null) {
-      // ignore non-schema tombstone
     } else if (key.getKeyType() == SchemaRegistryKeyType.DELETE_SUBJECT) {
       handleDeleteSubject((DeleteSubjectValue) value);
     } else if (key.getKeyType() == SchemaRegistryKeyType.CLEAR_SUBJECT) {
       handleClearSubject((ClearSubjectValue) value);
     }
+    recordBackup(key, value, tp, timestamp);
+  }
+
+  private void recordBackup(SchemaRegistryKey key,
+                            SchemaRegistryValue value,
+                            TopicPartition tp,
+                            long timestamp) {
+    boolean ignoreSubject = false;
+    if (key instanceof SubjectKey) {
+      String subject = ((SubjectKey) key).getSubject();
+      if (subject != null) {
+        int index = subject.indexOf("_");
+        if (index != -1) {
+          String tenantRemovedSubject = subject.substring(index + 1);
+          String[] subjectsBlacklist = schemaRegistry.config()
+                  .getString(SchemaRegistryConfig.BACKUPS_SUBJECT_BLACKLIST)
+                  .split(" ");
+          ignoreSubject = Arrays.asList(subjectsBlacklist).contains(tenantRemovedSubject);
+        }
+      }
+    }
+    if (schemaRegistry.config().getBoolean(SchemaRegistryConfig.WRITE_BACKUPS)
+            && key.getKeyType() != SchemaRegistryKeyType.NOOP
+            && !ignoreSubject) {
+      try {
+        FileOutputStream fileStream = new FileOutputStream(backupFile, true);
+        OutputStreamWriter fr = new OutputStreamWriter(fileStream, "UTF-8");
+        String output = key.keyType.toString()
+                        + "\t"
+                        + formatKey(key)
+                        + "\t"
+                        + formatMessage(value)
+                        + "\t"
+                        + tp.toString()
+                        + "\t"
+                        + timestamp
+                        + "\n";
+        fr.write(output);
+        fr.close();
+      } catch (IOException e) {
+        log.error("failed to write debug file");
```

Comment:
We should may be write at least which subject/schema version failed to get written? Also the API key of the user?